python/pyiceberg/io/pyarrow.py

```diff
@@ -555,8 +555,10 @@ def project_table(
 
     if len(tables) > 1:
         return pa.concat_tables(tables)
-    else:
+    elif len(tables) == 1:
         return tables[0]
+    else:
+        return pa.Table.from_batches([], schema=schema_to_pyarrow(projected_schema))
```

Comment:
Should this be in a separate PR?

Reply:
I can split this into a separate PR, but I think it makes sense to ship it with this PR since if all the datafiles are skipped, it will return an empty iterator that can't be concatenated.